src/matrixrtc/NewMembershipManager.ts

```diff
@@ -0,0 +1,808 @@
+                                ts: Date.now() + Math.max(retryIn, this.callMemberEventRetryDelayMinimum),
+                                type: MembershipActionType.Update,
+                            });
+                        },
+                        () => {
+                            throw Error(
+                                "Exceeded maximum own Membership state update attempts (client.sendStateEvent): " + e,
+                            );
+                        },
+                    );
+                    if (!rateLimitHandled) {
+                        this.scheduler.addAction({
+                            ts: Date.now() + this.callMemberEventRetryDelayMinimum,
+                            type: MembershipActionType.Update,
+                        });
+                    }
+                }
+                break;
+            case MembershipActionType.SendJoinEvent:
+                try {
+                    await this.client.sendStateEvent(
+                        this.room.roomId,
+                        EventType.GroupCallMemberPrefix,
+                        this.makeMyMembership(state.nextRelativeExpiry),
+                        this.stateKey,
+                    );
+                    state.nextRelativeExpiry += this.membershipEventExpiryTimeout;
+                    state.hasMemberStateEvent = true;
+                    state.sendMembershipRetries = 0;
+                    this.scheduler.addAction({ ts: Date.now(), type: DelayedLeaveActionType.RestartDelayedEvent });
+                    this.scheduler.addAction({
+                        ts: Date.now() + this.membershipTimerExpiryTimeout,
+                        type: MembershipActionType.Update,
+                    });
+                } catch (e) {
+                    this.handleRateLimitError(
+                        e,
+                        state.sendMembershipRetries,
+                        (resendDelay) => {
+                            state.sendMembershipRetries++;
+                            this.scheduler.addAction({
+                                ts: Date.now() + resendDelay,
+                                type: MembershipActionType.SendJoinEvent,
+                            });
+                        },
+                        () => {
+                            throw Error(
+                                "Exceeded maximum Own Membership state update attempts (client.sendStateEvent): " + e,
+                            );
+                        },
+                    );
+                }
+                break;
+            case MembershipActionType.SendLeaveEvent:
+                // We are good already
+                if (!state.hasMemberStateEvent) return;
+
+                // This is only a fallback in case we do not have working delayed events support.
+                // first we should try to just send the scheduled leave event
+                try {
+                    this.client.sendStateEvent(
```

Comment:
Missing `await`?

Reply:
I will look into the future lint pr today!